### ISSUE TITLE
What is the format of device ids?

### ISSUE THREAD

**Issue by pes10k** [w3c/mediacapture-screen-share#133]
Currently getUserMedia uses globally unique identifiers, which creates significant tracking risk

In the conversation around https://github.com/w3c/mediacapture-main/issues/607, it was stated that it would be difficult-to-impossible to remove these UUID device IDs from existing devices, but going forward UUIDs would no longer be used.  Could the spec authors discuss and specify that UUID-style device ids are no longer being used?

**Comment by dontcallmedom** [comment]
@snyderp can you clarify whether you meant this issue for this spec or for `getUserMedia`? In the Screen Capture API, deviceIds aren't exposed at all, so their format should not really matter one way or another.
>  Display capture sources therefore cannot be selected with the deviceId constraint, since their deviceIds are not exposed. 
https://www.w3.org/TR/2019/WD-screen-capture-20191119/#device-identifiers

**Comment by pes10k** [comment]
@dontcallmedom if i follow correctly, `getDisplayMedia` returns a `MediaStream`, and that `MediaStream` has an [`id`](https://www.w3.org/TR/mediacapture-streams/#dom-mediastream-id).

This issue is about whether that id is UUID-like (and so identifying / tracking risk), or whether, as per the convo in https://github.com/w3c/mediacapture-main/issues/607, its been changed to something not uniquely identifying

**Comment by dontcallmedom** [comment]
If the issue is about `MediaStream.id`, it should be filed in https://github.com/w3c/mediacapture-main/ since that is where the attribute is defined as a UUID, and the Screen Capture spec doesn't (and can't) change anything about it. I also don't think this particular id presents a fingerprinting risk since it is never persisted (contrarily to deviceId).

But the title of your issue suggests you were talking about deviceId, not MediaStream.id - and while there is a deviceId in Screen Capture, contrarily to Media Capture and Streams, it is not exposed to JavaScript and hence is not fingerprintable (no matter its format, which is not defined by the spec).

I could also be missing something, so I would appreciate your guidance.

**Comment by pes10k** [comment]
@dontcallmedom i see, then yes, maybe i put things in the wrong spot.  if so, apologies :)

But the reason i put it here is that the outcome of TPAC conversation with the media capture folks was that, for web compat reasons, the UUID-ness of MediaStream.id couldn't be changed for existing device types, but that for new uses of the API and new device-types, UUIDs would not be used.

So, this seemed the right place to specify that, possibly though WebIDL extension to MediaStream (i.e. `ScreenStream` or something) that just makes the non-uuid-ness explicit.  I didn't think https://github.com/w3c/mediacapture-main/ was the right place, since that would be the parent spec describing a change to a "child" spec, which (i though) would have been backwards.

TL;DR thats the problem, any nudging the right direction you could do would be appreciated :)

**Comment by youennf** [comment]
MediaDeviceInfo.deviceId may persist over browsing sessions and should be treated carefully as we are currently discussing.

MediaStream IDs are not persistent, they last as long as the MediaStream last.
In that sense, they do not pause a privacy threat.

**Comment by pes10k** [comment]
Ah, terrific then, my misunderstanding.  As long as this spec does not involve the creation of any site/JS-accessible UUIDs, then my concern is addressed and I'll close the issue.

Thanks to you both for the handholding!
